File: goblobBase/src/main/java/com/cauchymop/goblob/model/BoardViewModel.java
package com.cauchymop.goblob.model;

import com.cauchymop.goblob.proto.PlayGameData;

public class BoardViewModel {

  private final int boardSize;
  private PlayGameData.Color[][] stones;
  private PlayGameData.Color[][] territories;
  private int lastMoveX;
  private int lastMoveY;
  private boolean isInteractive;

  public BoardViewModel(int boardSize, PlayGameData.Color[][] stones,
      PlayGameData.Color[][] territories, int lastMoveX, int LastMoveY, boolean isInteractive) {
    this.boardSize = boardSize;
    this.stones = stones;
    this.territories = territories;
    this.lastMoveX = lastMoveX;
    this.lastMoveY = LastMoveY;
    this.isInteractive = isInteractive;
  }

  public int getBoardSize() {
    return boardSize;
  }

  public PlayGameData.Color getColor(int x, int y) {
    return stones[y][x];
  }

  public PlayGameData.Color getTerritory(int x, int y) {
    return territories[y][x];
  }

  public boolean isLastMove(int x, int y) {
    return (x == lastMoveX) && (y == lastMoveY);
  }

  public boolean isInteractive() {
    return isInteractive;
  }
}

File: goblobBase/src/main/java/com/cauchymop/goblob/viewmodel/BoardViewModel.kt
package com.cauchymop.goblob.viewmodel

import com.cauchymop.goblob.proto.PlayGameData

data class BoardViewModel(
        val boardSize: Int,
        private val stones: Array<Array<PlayGameData.Color?>>,
        private val territories: Array<Array<PlayGameData.Color?>>,
        private val lastMoveX: Int,
        private val lastMoveY: Int,
        val isInteractive: Boolean) {

    fun getColor(x: Int, y: Int) = stones[y][x]

    fun getTerritory(x: Int, y: Int) = territories[y][x]

    fun isLastMove(x: Int, y: Int) = (x == lastMoveX && y == lastMoveY)

}
